<a href="https://colab.research.google.com/github/Satyake/Shopily-Challenge-Link/blob/main/Shopify_Challenge_Link.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [29]:
import pandas as pd 

In [30]:
data=pd.read_csv('/content/2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv')

In [31]:
data.head()

order_id  shop_id  user_id  ...  total_items  payment_method           created_at
0         1       53      746  ...            2            cash  2017-03-13 12:36:56
1         2       92      925  ...            1            cash  2017-03-03 17:38:52
2         3       44      861  ...            1            cash   2017-03-14 4:23:56
3         4       18      935  ...            1     credit_card  2017-03-26 12:43:37
4         5       18      883  ...            1     credit_card   2017-03-01 4:35:11

[5 rows x 7 columns]

In [32]:
data.shape

(5000, 7)

In [33]:
data.describe()


order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

In [38]:
order_amount=data.iloc[:,3].values
total_items=data.iloc[:,4].values


In [45]:
total_items

array([2, 1, 1, ..., 3, 2, 2])

In [53]:
np.mean(order_amount)

3145.128

In [41]:
sumed=np.sum(total_items)

In [43]:
order_amount_sum=np.sum(order_amount)

In [46]:
total_amounts_sum=np.sum(total_items)


In [49]:
order_amount_sum/total_amounts_sum

357.92152221412965

In [59]:
np.median(order_amount)

284.0

357.9 is the average! 

The following was the issue: , The Wrong AOV value was due to not taking into account the number of orders, instead it simply gave the mean of the amounts and not the actual mean amount. The order_amount needs to be divided by the total_amounts.

Median would be a better metric, in this case as we can see , the most common  amount orders.